src/explore-education-statistics-admin/src/components/comments/CommentsWrapper.tsx

```diff
@@ -0,0 +1,101 @@
+import styles from '@admin/components/comments/CommentsWrapper.module.scss';
+import CommentAddForm from '@admin/components/comments/CommentAddForm';
+import CommentsList from '@admin/components/comments/CommentsList';
+import { useCommentsContext } from '@admin/contexts/CommentsContext';
+import React, { ReactNode, useRef } from 'react';
+import { CommentType } from 'src/services/types/content';
+import classNames from 'classnames';
+import { Toggle } from '@common/hooks/useToggle';
+import Button from '@common/components/Button';
+import Tooltip from '@common/components/Tooltip';
+
+interface Props {
+  allowComments?: boolean;
+  blockLockedMessage?: string;
+  children: ReactNode;
+  commentType: CommentType;
+  id: string;
+  showCommentAddForm?: boolean;
+  showCommentsList?: boolean;
+  testId?: string;
+  toggleCommentAddForm?: Toggle;
+  onViewComments?: () => void;
+}
+
+export default function CommentsWrapper({
+  allowComments = true,
+  blockLockedMessage,
+  children,
+  id,
+  commentType,
+  showCommentAddForm,
+  showCommentsList = true,
+  testId,
+  toggleCommentAddForm,
+  onViewComments,
+}: Props) {
+  const containerRef = useRef<HTMLDivElement>(null);
+  const { comments } = useCommentsContext();
+  const isInline = commentType === 'inline';
+
+  return (
+    <div className={styles.container} ref={containerRef} data-testid={testId}>
+      {allowComments && (
+        <div className={styles.sidebar} data-testid="comments-sidebar">
+          {showCommentAddForm && toggleCommentAddForm && (
+            <CommentAddForm
+              baseId={id}
+              containerRef={containerRef}
+              onCancel={toggleCommentAddForm?.off}
+              onSave={toggleCommentAddForm?.off}
+            />
+          )}
+          {!showCommentAddForm && !isInline && (
+            <Button
+              className="govuk-!-width-full govuk-!-margin-bottom-2"
+              testId="comment-add-button"
+              variant="secondary"
+              onClick={toggleCommentAddForm?.on}
+            >
+              Add comment
+            </Button>
+          )}
+
+          {!showCommentsList && comments.length > 0 && (
+            <Tooltip text={blockLockedMessage} enabled={!!blockLockedMessage}>
+              {({ ref }) => (
+                <Button
+                  ariaDisabled={!!blockLockedMessage}
+                  className="govuk-!-width-full"
+                  ref={ref}
+                  testId="view-comments"
+                  variant="secondary"
+                  onClick={onViewComments}
+                >
+                  View comments
+                  <br />
+                  <span className="govuk-!-margin-top-1 govuk-body-s">
+                    {`(${
+                      comments.filter(comment => !comment.resolved).length
+                    } unresolved)`}
+                  </span>
+                </Button>
+              )}
+            </Tooltip>
+          )}
```

Comment:
Kinda forgotten the rationale for this, but why do content blocks need to toggle to editing mode when this button is clicked?

It's a little janky and I've suggested in another comment that we could use details to hide comments. It'd be good if we could actually display these in the details contents, but this doesn't work so well if we're having to toggle to editing mode as well.

If the current implementation is because we're relying on bits of CKEditor functionality to highlight / focus, I can't really see why we couldn't just implement this ourselves?

The markers are in the HTML, so it looks like it'd be possible to get this working without CKEditor in the read-only view.

Reply:
Yes it's because the markers are tied into CKEditor. It's not just for highlighting, there's also handling resolving/unresolving and deleting comments. It would be possible to change it so this could all be handled in read only mode as well as with the editor open but it wouldn't be a trivial change and I think it'd be outside the scope of this PR.